include/revng/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,240 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <concepts>
+
+#include "llvm/ADT/ArrayRef.h"
+
+#include "revng/ABI/Specializations/AArch64.h"
+#include "revng/ABI/Specializations/ARM.h"
+#include "revng/ABI/Specializations/Microsoft_x86_64.h"
+#include "revng/ABI/Specializations/Microsoft_x86_64_clrcall.h"
+#include "revng/ABI/Specializations/Microsoft_x86_64_vectorcall.h"
+#include "revng/ABI/Specializations/Microsoft_x86_cdecl.h"
+#include "revng/ABI/Specializations/Microsoft_x86_clrcall.h"
+#include "revng/ABI/Specializations/Microsoft_x86_fastcall.h"
+#include "revng/ABI/Specializations/Microsoft_x86_stdcall.h"
+#include "revng/ABI/Specializations/Microsoft_x86_thiscall.h"
+#include "revng/ABI/Specializations/Microsoft_x86_vectorcall.h"
+#include "revng/ABI/Specializations/Pascal_x86.h"
+#include "revng/ABI/Specializations/SystemV_MIPS_o32.h"
+#include "revng/ABI/Specializations/SystemV_x86.h"
+#include "revng/ABI/Specializations/SystemV_x86_64.h"
+#include "revng/ABI/Specializations/SystemV_x86_regparm_1.h"
+#include "revng/ABI/Specializations/SystemV_x86_regparm_2.h"
+#include "revng/ABI/Specializations/SystemV_x86_regparm_3.h"
+#include "revng/ABI/Specializations/SystemZ_s390x.h"
```

Comment:
```suggestion
#include "revng/ABI/AArch64/AArch64.h"
#include "revng/ABI/ARM/ARM.h"
#include "revng/ABI/MIPS/SystemV_MIPS_o32.h"
#include "revng/ABI/x86/Microsoft_x86_cdecl.h"
#include "revng/ABI/x86/Microsoft_x86_clrcall.h"
#include "revng/ABI/x86/Microsoft_x86_fastcall.h"
#include "revng/ABI/x86/Microsoft_x86_stdcall.h"
#include "revng/ABI/x86/Microsoft_x86_thiscall.h"
#include "revng/ABI/x86/Microsoft_x86_vectorcall.h"
#include "revng/ABI/x86/Pascal_x86.h"
#include "revng/ABI/x86/SystemV_x86.h"
#include "revng/ABI/x86/SystemV_x86_regparm_1.h"
#include "revng/ABI/x86/SystemV_x86_regparm_2.h"
#include "revng/ABI/x86/SystemV_x86_regparm_3.h"
#include "revng/ABI/x86_64/SystemV_x86_64.h"
#include "revng/ABI/x86_64/Microsoft_x86_64.h"
#include "revng/ABI/x86_64/Microsoft_x86_64_clrcall.h"
#include "revng/ABI/x86_64/Microsoft_x86_64_vectorcall.h"
#include "revng/ABI/s390x/SystemZ_s390x.h"
```

Aren't the ARM and AArch64 calling convention SystemV? I might already have asked this.

Reply:
Well, neither doc explicitly mentions that, so I was reluctant to force the naming - they look pretty different to me.

To be extra specific, it does mention `SysV` on page 11 of the doc when talking about types. But that's about it.

Reply:
I think the formal name would be `AAPCS64`:

> The AAPCS64 defines how subroutines (Page 11)

AFAIU it stands for "Arm 64-bit Architecture Procedure Call Standard".
But yeah, if it's not explicitly SystemV, we can leave it as it is.

Does Microsoft adopts the same calling convention?

Reply:
What do you mean by "the same"? As far as I understand, Microsoft naming convention is "non-microsoft ABIs don't even exist", so they are named simply `x64`, `stdcall`, `vectorcall`, etc.

Reply:
And also, as far as I understand `AAPCS64` is not really the name - just an abbreviation (it's listed in the relevant part of the document, page 9)

Reply:
OK, but if the full name is "Arm 64-bit Architecture Procedure Call Standard" it's a bit too long and the `AAPCS64` abbreviation could work.
The document is sprayed all over with `AAPCS64`.
![image](https://user-images.githubusercontent.com/2545644/153564295-3aaee106-beda-4147-9476-5edc7750ea6e.png)
![image](https://user-images.githubusercontent.com/2545644/153564258-e2b7957d-59c4-47d4-b59a-54c5abb1c801.png)

Reply:
Done and done!